CollapseLauncher/XAMLs/MainApp/Pages/SettingsPage.xaml.cs

```diff
@@ -1328,6 +1329,118 @@ private double DownloadChunkSize
             }
         }
 #nullable restore
+
+        #region Database
+
+        private bool IsDbEnabled
+        {
+            get => DbHandler.IsEnabled;
+            set
+            {
+                DbHandler.IsEnabled = value;
+                if (value) _ = DbHandler.Init();
+            }
+        }
+
+        private string DbUrl
+        {
+            get => DbHandler.Uri;
+            set
+            {
+                // Automatically replace libsql protocol to https
+                if (value.Contains("libsql://", StringComparison.InvariantCultureIgnoreCase))
+                    value = value.Replace("libsql", "https");
+                if (!value.Contains("https://"))
+                {
+                    DbUriTextBox.Text = DbHandler.Uri;
+                    return;
+                }
+
+                DbHandler.Uri = value;
+            }
+        }
+
+        private string DbToken
+        {
+            get => DbHandler.Token;
+            set => DbHandler.Token = value;
+        }
+
+        private string _currentDbGuid;
+        private string DbUserId
+        {
+            get => DbHandler.UserId.ToString();
+            set
+            {
+                if (Guid.TryParse(value, out var guid))
+                {
+                    DbHandler.UserId = guid;
+                }
+                else
+                {
+                    DbUserIdTextBox.Text = DbHandler.UserId.ToString();
+                }
+            }
+        }
+
+        private void DbUserIdTextBox_GotFocus(object sender, RoutedEventArgs e)
+        {
+            _currentDbGuid = DbUserIdTextBox.Text;
+        }
+
+        private async void DbUserIdTextBox_LostFocus(object sender, RoutedEventArgs e)
```

Comment:
After changing the ID, the confirmation dialog will appear when clicking any other button/toggle, the click action won't be executed in the case of the button but it will if its a toggle.

This can create a bit of user confusion.

Reply:
I'm not sure on how to fix this behavior since this seems to be another WinUI shenanigans

Reply:
Resolving since it seems to be an issue with WinUI's implementation.